src/ALE/Recon1d_PPM_CW.F90

```diff
@@ -0,0 +1,418 @@
+  real :: d12 ! h1 + h2  but used in the denominator so include h_neglect [H]
+  real :: h01_h112, h23_h122 ! Approximately 2/3 [nondim]
+  real :: ddh ! Approximately 0 [nondim]
+  real :: I_h12, I_h0123 ! Reciprocals of d12 and sum(h) [H-1]
+  real :: dul, dur ! Left and right cell PLM slopes [A]
+  real :: u0, u1, u2 ! Far left, left, and right cell values [A]
+  real :: edge ! Edge value between cell k-1 and k [A]
+  real :: u_min, u_max ! Minimum and maximum value across edge [A]
+  real :: a6 ! Colella and Woodward curvature [A]
+  real :: du ! Difference between edges across cell [A]
+  real :: slp(this%n) ! PLM slope [A]
+  integer :: k, n
+
+  n = this%n
+
+  ! First populate the PLM reconstructions
+  call this%PLM%reconstruct( h, u )
+  do k = 1, n
+    slp(k) = this%PLM%ur(k) - this%PLM%ul(k)
+  enddo
+  ! Extrapolate from interior for boundary PLM slopes
+  ! Note: this is not conventional but helps retain accuracy near top/bottom
+  ! boudaries and reduces the adverse influence of the boudnaries int he interior
+  ! reconstructions. The final PPM reconstruction is still bounded to PCM.
+  slp(1) = 2.0 * ( this%PLM%ul(2) - u(1) )
+  slp(n) = 2.0 * ( u(n) - this%PLM%ur(n-1) )
+
+  do K = 2, n ! K=2 is interface between cells 1 and 2
+    h0 = h( max( 1, k-2 ) ) ! This treatment implies a virtual mirror cell at k=0
+    h1 = h(k-1)
+    h2 = h(k)
+    h3 = h( min( n, k+1 ) ) ! This treatment implies a virtual mirror cell at k=n+1
+    d12 = ( h1 + h2 ) + this%h_neglect    ! d12 is only ever used in the denominator
+    h01_h112 = ( h0 + h1 ) / ( h1 + d12 ) ! When uniform -> 2/3
+    h23_h122 = ( h2 + h3 ) / ( d12 + h2 ) ! When uniform -> 2/3
+    ddh = h01_h112 - h23_h122             ! When uniform -> 0
+    I_h12 = 1.0 / d12                     ! When uniform -> 1/(2h)
+    I_h0123 = 1.0 / ( d12 + ( h0 + h3 ) ) ! When uniform -> 1/(4h)
+    dul = slp(k-1)
+    dur = slp(k)
+    u2 = u(k)
+    u1 = u(k-1)
+    edge = I_h12 * ( h2 * u1 + h1 * u2 ) &                              ! 1/2 u1 + 1/2 u2
+         + I_h0123 * ( 2.0 * h1 * h2 * I_h12 * ddh * ( u2 - u1 ) &      ! 0
+                     + ( h2 * h23_h122 * dul - h1 * h01_h112 * dur ) )  ! 1/6 dul - 1/6 dur
+    u_min = min( u1, u2 )
+    u_max = max( u1, u2 )
+    edge = max( min( edge, u_max), u_min ) ! Unclear if we need this bounding in the interior
+    this%ur(k-1) = edge
+    this%ul(k) = edge
+  enddo
+  this%ul(1) = u(1) ! PCM
+  this%ur(1) = u(1) ! PCM
+  this%ur(n) = u(n) ! PCM
+  this%ul(n) = u(n) ! PCM
+
+  do K = 2, n ! K=2 is interface between cells 1 and 2
+    u0 = u(k-1)
+    u1 = u(k)
+    u2 = u(k+1)
+    a6 = 3.0 * ( ( u1 - this%ul(k) ) + ( u1 - this%ur(k) ) )
+    du = this%ur(k) - this%ul(k)
+    if ( ( u2 - u1 ) * ( u1 - u0 ) <- 0.0 ) then ! Large scale extrema
+      this%ul(k) = u1
+      this%ur(k) = u1
+    elseif ( du * a6 > du * du ) then ! Extrema on right
+      edge = 3.0 * u1 - 2.0 * this%ur(k) ! Subject to round off
```

Comment:
Why is this expression for edge, which is never used and is replaced in the next line, not commented out?  A similar comment would apply to line 173.

Reply:
Commented out, plus added more comments about following commented-out expressions